fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java

```diff
@@ -308,19 +308,19 @@ public static void analyzeTableRef(TableRef tableRef, String dbName, Database db
             }
         }
 
-        OlapTable olapTbl = null;
+
         if (tbl instanceof OlapTable) {
-            olapTbl = (OlapTable) tbl;
-            PartitionInfo partitionInfo = olapTbl.getPartitionInfo();
+            PartitionInfo partitionInfo = ((OlapTable) tbl).getPartitionInfo();
             if (partitionInfo instanceof ListPartitionInfo) {
                 throw new SemanticException("List partition table does not support backup/restore job");
             }
         }
 
-        if (olapTbl != null && partitionNames != null) {
+        if (partitionNames != null) {
             if (!tbl.isOlapOrCloudNativeTable()) {
                 ErrorReport.reportSemanticException(ErrorCode.ERR_WRONG_TABLE_NAME, tableName.getTbl());
             }
+            OlapTable olapTbl = (OlapTable) tbl;
```

Comment:
Does it work well when tbl is a cloud-native table?

Reply:
Yes, cloud-native table is also olap table